### PR TITLE
Added User Entity to book-store-app:

### PR DESCRIPTION
**Pull request by GoldHolub**
- added User entity
- added the required DTOs and mapper
- added data validation in the new DTO classes
- created FieldMatch annotation for checking that password and repeatPassword field are the same
- GlobalExceptionHandle handle the RegistrationException
- added soft delete of users

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -72,7 +72,7 @@
 			<version>${mapstruct.version}</version>
 		</dependency>
 
-		<!-- SWAGGER -->
+		<!-- OPEN.API -->
```
Comment (Rommelua) [inline pom.xml]: remove this comment

File: src/main/java/com/example/demo/annotation/FieldMatchValidator.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        Object firstValue = new BeanWrapperImpl(value).getPropertyValue(firstFieldName);
+        Object secondValue = new BeanWrapperImpl(value).getPropertyValue(secondFieldName);
+        if (firstValue != null) {
+            return firstValue.equals(secondValue);
+        } else {
+            return secondValue == null;
+        }
```
Comment (Rommelua) [inline src/main/java/com/example/demo/annotation/FieldMatchValidator.java]:
```suggestion
            return Objects.equals(firstValue, secondValue);
```

File: src/main/java/com/example/demo/dto/user/UserRegistrationRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+@Validated
+public class UserRegistrationRequestDto {
+    @NotBlank
+    @NotNull
+    private String email;
+    @NotBlank
+    @NotNull
+    private String password;
+    @NotBlank
+    @NotNull
```
Comment (Rommelua) [inline src/main/java/com/example/demo/dto/user/UserRegistrationRequestDto.java]:
```suggestion
```

File: src/main/java/com/example/demo/dto/user/UserRegistrationRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+@FieldMatch.List({
+        @FieldMatch(first = "password", second = "repeatPassword",
+                    message = "The password fields must match")
+})
+@Data
+@Validated
+public class UserRegistrationRequestDto {
+    @NotBlank
+    @NotNull
+    private String email;
```
Comment (Rommelua) [inline src/main/java/com/example/demo/dto/user/UserRegistrationRequestDto.java]:
```suggestion
    @NotNull
    @Email
    private String email;
```

File: src/main/java/com/example/demo/dto/user/UserRegistrationRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    @NotBlank
+    @NotNull
```
Comment (Rommelua) [inline src/main/java/com/example/demo/dto/user/UserRegistrationRequestDto.java]: image

File: src/main/java/com/example/demo/controller/AuthenticationController.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+@Validated
+@RestController
+public class AuthenticationController {
+    private final UserService userService;
+
+    @Operation(summary = "User registration",
+            description = "register a new user")
+    @PostMapping(value = "/registration")
+    public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto requestDto)
+                                            throws RegistrationException {
```
Comment (sokimaaa) [inline src/main/java/com/example/demo/controller/AuthenticationController.java:29]:
```suggestion
                                         {
```

File: src/main/java/com/example/demo/exception/RegistrationException.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package com.example.demo.exception;
+
+public class RegistrationException extends Exception {
```
Comment (sokimaaa) [inline src/main/java/com/example/demo/exception/RegistrationException.java:3]: make unchecked exception

File: src/main/java/com/example/demo/service/UserService.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package com.example.demo.service;
+
+import com.example.demo.dto.user.UserRegistrationRequestDto;
+import com.example.demo.dto.user.UserResponseDto;
+import com.example.demo.exception.RegistrationException;
+
+public interface UserService {
+    UserResponseDto register(UserRegistrationRequestDto requestDto) throws RegistrationException;
```
Comment (sokimaaa) [inline src/main/java/com/example/demo/service/UserService.java:8]:
```suggestion
    UserResponseDto register(UserRegistrationRequestDto requestDto);
```

**Pull request merged**